provider/pkg/provider/environment.go

```diff
@@ -238,10 +238,14 @@ func (st *PulumiServiceEnvironmentResource) Check(req *pulumirpc.CheckRequest) (
 		}
 	}
 
-	yamlBytes, err := getBytesFromAsset(inputMap["yaml"].AssetValue())
-	if err != nil {
-		return nil, err
+	yamlBytes := []byte{}
```

Comment:
Hmm... does this mean there will be no `yamlBytes` if it's computed? Or does this only affect `Check` and the resolved value of environment.yaml is still correct?

Reply:
This only affects `Check`. I have confirmed the resolved value is still correct (I'm assuming that it's ok to not have computed values during Check?)

Reply:
The purpose of this Check is to take the string value of the yaml and put that into state, so there's always a string value to compare against. In this case, when it's computed, it'll result in yaml being empty in state. So next time there's an update, there will be a diff because there is no value for `yaml`. You can test this by adding an example test similar to in https://github.com/pulumi/pulumi-pulumiservice/pull/391/files

We'll need a way to get the string value even out of the computed property I think.

Reply:
Computed/unknown values will only appear during a preview, so I think that this will be okay. We'll re-run the `Check` during the update, at which point all values will be known.

Reply:
Though I do think I'd do this slightly differently--instead of checking `IsAsset` I'd check for `!IsComputed`.